nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -681,57 +254,19 @@ contract StakingEscrow is Issuer, IERC900History {
         uint256 _value,
         uint16 _unlockingDuration
     )
-        external
+        external onlyWorkLock
     {
-        require(msg.sender == address(workLock));
+        require(_value != 0, "Amount of tokens to deposit must be specified");
         StakerInfo storage info = stakerInfo[_staker];
-        if (!info.flags.bitSet(WIND_DOWN_INDEX) && info.subStakes.length == 0) {
-            info.flags = info.flags.toggleBit(WIND_DOWN_INDEX);
-            emit WindDownSet(_staker, true);
-        }
-        // WorkLock still uses the genesis period length (24h)
-        _unlockingDuration = recalculatePeriod(_unlockingDuration);
-        deposit(_staker, msg.sender, MAX_SUB_STAKES, _value, _unlockingDuration);
-    }
-
-    /**
-    * @notice Set `windDown` parameter.
-    * If true then stake's duration will be decreasing in each period with `commitToNextPeriod()`
-    * @param _windDown Value for parameter
-    */
-    function setWindDown(bool _windDown) external {
-        StakerInfo storage info = stakerInfo[msg.sender];
-        if (info.flags.bitSet(WIND_DOWN_INDEX) == _windDown) {
-            return;
-        }
-        info.flags = info.flags.toggleBit(WIND_DOWN_INDEX);
-        emit WindDownSet(msg.sender, _windDown);
-
-        // duration adjustment if next period is committed
-        uint16 nextPeriod = getCurrentPeriod() + 1;
-        if (info.nextCommittedPeriod != nextPeriod) {
-           return;
+        // initial stake of the staker
+        if (info.value == 0 && info.lastCommittedPeriod == 0) {
+            stakers.push(_staker);
         }
+        token.safeTransferFrom(msg.sender, address(this), _value);
+        info.value += _value;
 
-        // adjust sub-stakes duration for the new value of winding down parameter
-        for (uint256 index = 0; index < info.subStakes.length; index++) {
-            SubStakeInfo storage subStake = info.subStakes[index];
-            // sub-stake does not have fixed last period when winding down is disabled
-            if (!_windDown && subStake.lastPeriod == nextPeriod) {
-                subStake.lastPeriod = 0;
-                subStake.unlockingDuration = 1;
-                continue;
-            }
-            // this sub-stake is no longer affected by winding down parameter
-            if (subStake.lastPeriod != 0 || subStake.unlockingDuration == 0) {
-                continue;
-            }
-
-            subStake.unlockingDuration = _windDown ? subStake.unlockingDuration - 1 : subStake.unlockingDuration + 1;
-            if (subStake.unlockingDuration == 0) {
-                subStake.lastPeriod = nextPeriod;
-            }
-        }
+        addSnapshot(info, int256(_value));
+        emit Deposited(_staker, _value);
```

Comment:
So, we are removing the possibility of depositing NU to StakingEscrow, but we're still allowing WorkLock deposits. Can this have implications on the Threshold side? Maybe a possibility is to somehow end WL completely (i.e. refund & no more deposits)?

Reply:
there is no any way to stop `WorkLock`, so I can't see how to avoid deposit. Re Threshold: all these "new" tokens are in `WorkLock` contract and they should be part of `currentPeriodSupply`. For me it looks like we already consider them as deposited in `StakingEscrow`

Reply:
> all these "new" tokens are in WorkLock contract and they should be part of currentPeriodSupply. For me it looks like we already consider them as deposited in StakingEscrow

OK, this is a good reasoning.

Reply:
No bids after Sep 28th 2020 (https://dune.xyz/queries/271579) and only one deposit on Aug 30th 2020 (https://dune.xyz/queries/271593). Would the only implication be that someone loses money for (inexplicably) sending in ETH and there's no longer away to call `worktoETH`?

Reply:
I don't understand the question. Both events is only for initial worklock phase which was more than year ago

Reply:
Yep, that's what I was getting at with my question – what exactly is the undesirable corner case where someone makes a deposit post initial worklock phase? Has that occurred in the last 13 months?

Reply:
This deposit for WorkLock, not for StakingEscrow